clients/client/src/main/java/org/projectnessie/client/api/PagingBuilder.java

```diff
@@ -45,24 +45,24 @@
    * with default page-size-hint and no limit on the number of returned entries.
    */
   default Stream<ENTRY> stream() throws NessieNotFoundException {
-    return stream(OptionalInt.empty(), OptionalInt.empty());
+    return stream(OptionalInt.empty());
   }
 
   /**
    * This is a convenience function that is equivalent to {@link #stream(OptionalInt, OptionalInt)}
    * with the specified page-size-hint and no limit on the number of returned entries.
    */
-  default Stream<ENTRY> stream(OptionalInt pageSizeHint) throws NessieNotFoundException {
-    return stream(pageSizeHint, OptionalInt.empty());
+  default Stream<ENTRY> stream(OptionalInt maxTotalRecords) throws NessieNotFoundException {
+    return stream(maxTotalRecords, OptionalInt.empty());
   }
 
   /**
    * Retrieve entries/results as a Java {@link Stream}, uses automatic paging.
    *
+   * @param maxTotalRecords {@link OptionalInt#empty()} for unlimited amount of results or a limit
    * @param pageSizeHint {@link OptionalInt#empty()} to let the Nessie server choose the page size,
    *     or provide a <em>proposal</em> to the Nessie server
-   * @param maxTotalRecords {@link OptionalInt#empty()} for unlimited amount of results or a limit
    */
-  Stream<ENTRY> stream(OptionalInt pageSizeHint, OptionalInt maxTotalRecords)
+  Stream<ENTRY> stream(OptionalInt maxTotalRecords, OptionalInt pageSizeHint)
```

Comment:
This could be confusing to end users too, I guess. WDYT about moving `maxTotalRecords` to a builder method?

Reply:
We already have `maxRecords`, which is actually the page size. I tend to remove the `pageSizeHint` option - if people really need finer grained control, we can revisit this.